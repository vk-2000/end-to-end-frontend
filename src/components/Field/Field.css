.content-fields-body-item {
    width: 100%;
    height: 50px;
    background-color: white;
    border-radius: 10px;
    margin: 10px 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 20px;
}