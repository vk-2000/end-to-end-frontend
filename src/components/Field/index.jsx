import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Field.css';
import deleteIcon from '../../assets/icons/delete-icon.png';
import editIcon from '../../assets/icons/user-edit-large.png';

const Field = ({
  field, type, handleFieldDelete, handleFieldUpdate,
}) => {
  const [showInputEle, setShowInputEle] = useState(false);
  const handleKeyDown = (e) => {
    if (e.key === 'Enter' && e.target.value !== '') {
      handleFieldUpdate(field, e.target.value);
      setShowInputEle(false);
    }
  };
  return (
    <div className="content-fields-body-item">
      <div className="field-image">
        Ab
      </div>
      <div className="field-container">
        <div className="field-name">
          {showInputEle ? (
            <input defaultValue={field} onKeyDown={handleKeyDown} />
          ) : (
            <span>{field}</span>
          )}
        </div>
        <div className="field-type">
          {type}
        </div>
        <div className="edit-actions-field">
          <button className="btn-edit" onClick={() => setShowInputEle(true)} type="button">
            <img src={editIcon} alt="edit" />
          </button>
          <button className="btn-delete" onClick={() => handleFieldDelete(field)} type="button">
            <img src={deleteIcon} alt="delete" />
          </button>
        </div>
      </div>
    </div>
  );
};

export default Field;

Field.propTypes = {
  field: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  handleFieldDelete: PropTypes.func.isRequired,
  handleFieldUpdate: PropTypes.func.isRequired,
};
