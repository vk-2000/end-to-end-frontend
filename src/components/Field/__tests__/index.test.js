import { fireEvent, render } from '@testing-library/react';
import Field from '..';

describe('Field', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Field
      field="Hello"
      type="World"
      handleFieldDelete={jest.fn()}
      handleFieldUpdate={jest.fn()}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should show input when clicked', () => {
    const { getByTestId } = render(<Field
      field="Hello"
      type="World"
      handleFieldDelete={jest.fn()}
      handleFieldUpdate={jest.fn()}
    />);
    const btnEdit = getByTestId('btn-field-edit');
    fireEvent.click(btnEdit);
    expect(getByTestId('field-test-input')).toBeInTheDocument();
  });
  it('should call handleFieldUpdate when input is changed and enter is pressed', () => {
    const handleFieldUpdate = jest.fn();
    const { getByTestId } = render(<Field
      field="Hello"
      type="World"
      handleFieldDelete={jest.fn()}
      handleFieldUpdate={handleFieldUpdate}
    />);
    const btnEdit = getByTestId('btn-field-edit');
    fireEvent.click(btnEdit);
    const input = getByTestId('field-test-input');
    fireEvent.change(input, { target: { value: 'Hello World' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 'Enter' });
    expect(handleFieldUpdate).toHaveBeenCalled();
  });
  it('should not call handleFieldUpdate when input is changed and outside is pressed', () => {
    const handleFieldUpdate = jest.fn();
    const { getByTestId } = render(<Field
      field="Hello"
      type="World"
      handleFieldDelete={jest.fn()}
      handleFieldUpdate={handleFieldUpdate}
    />);
    const btnEdit = getByTestId('btn-field-edit');
    fireEvent.click(btnEdit);
    const input = getByTestId('field-test-input');
    fireEvent.change(input, { target: { value: 'Hello World' } });
    fireEvent.blur(input);
    expect(handleFieldUpdate).toHaveBeenCalled();
  });
  it('should call handleFieldDelete when delete button is clicked', () => {
    const handleFieldDelete = jest.fn();
    const { getByTestId } = render(<Field
      field="Hello"
      type="World"
      handleFieldDelete={handleFieldDelete}
      handleFieldUpdate={jest.fn()}
    />);
    const btnDelete = getByTestId('btn-field-delete');
    fireEvent.click(btnDelete);
    expect(handleFieldDelete).toHaveBeenCalled();
  });
});
