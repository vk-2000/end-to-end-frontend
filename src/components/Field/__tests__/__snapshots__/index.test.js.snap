// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should render correctly 1`] = `
<DocumentFragment>
  <div
    class="content-fields-body-item"
  >
    <div
      class="field-image"
    >
      Ab
    </div>
    <div
      class="field-container"
    >
      <div
        class="field-name"
      >
        <span>
          Hello
        </span>
      </div>
      <div
        class="field-type"
      >
        World
      </div>
      <div
        class="edit-actions-field"
      >
        <button
          class="btn-edit"
          data-testid="btn-field-edit"
          type="button"
        >
          <img
            alt="edit"
            src="user-edit-large.png"
          />
        </button>
        <button
          class="btn-delete"
          data-testid="btn-field-delete"
          type="button"
        >
          <img
            alt="delete"
            src="delete-icon.png"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
